src/test/scala/com/grandata/commons/geo/GeoLocatorSpec.scala

```diff
@@ -21,26 +20,18 @@ class GeoLocatorSpec extends Specification with BeforeAfterAll {
     }
   
     "return none for the points not located" in {
-      val result = new GeoLocator(List(sFile.getAbsolutePath, cFile.getAbsolutePath)).generateTrees.locate(GeoPoint(15.74,103.5))
+      val result = new GeoLocator(Seq(statesContent, citiesContent)).generateTrees.locate(GeoPoint(15.74,103.5))
       result must be size(2)
       result(0) must beSome
       result(0).get.get("id") === 5
       result(1) must beNone
     }
+    
+    "fix invalid geometries before building the R-tree" in {
+      val result = new GeoLocator(Seq(statesContent, citiesContent)).generateTrees.locate(GeoPoint(26.1, -109.1))
+      result must be size(2)
+      result(1).get.get("id") === 60
+
+    }
```

Comment:
It would be a good idea to add examples when Geolocator is used with fixGeometries=false.
Or if it will be used only with fixGeometries=true, maybe this parameter is dont needed.